modules/nwtc-library/src/NWTC_C_Binding.f90

```diff
@@ -49,7 +49,73 @@ SUBROUTINE SetErr(ErrStat, ErrMsg, ErrStat_C, ErrMsg_C)
     else
         ErrMsg_C = TRANSFER( TRIM(ErrMsg)//C_NULL_CHAR, ErrMsg_C )
     endif
-END SUBROUTINE SetErr
+END SUBROUTINE SetErrStat_F2C
+
+!> This subroutine incorporates the local error status and error messages into the global error
+!! status and message. It expects both local and global error messages to be null-terminated
+!! C strings.
+SUBROUTINE SetErrStat_C(ErrStatLocal, ErrMessLocal, ErrStatGlobal, ErrMessGlobal, RoutineName)
+
+    INTEGER(C_INT),                          INTENT(IN   ) :: ErrStatLocal                  ! Error status of the operation
+    CHARACTER(KIND=C_CHAR, LEN=ErrMsgLen_C), INTENT(IN   ) :: ErrMessLocal                  ! Error message if ErrStat /= ErrID_None
+    INTEGER(C_INT),                          INTENT(INOUT) :: ErrStatGlobal                 ! Error status of the operation
+    CHARACTER(KIND=C_CHAR),                  INTENT(INOUT) :: ErrMessGlobal(ErrMsgLen_C)    ! Error message if ErrStat /= ErrID_None
+    CHARACTER(*),                            INTENT(IN   ) :: RoutineName                   ! Name of the routine error occurred in
+
+    IF ( ErrStatLocal == ErrID_None ) RETURN
+
+    IF (ErrStatGlobal /= ErrID_None) THEN
+        ! print *, "in if", ErrStatGlobal, ErrID_None
+        ! ErrMessGlobal = TRIM(ErrMessGlobal)//new_line('a')
+        ! print *, "ErrMessGlobal", ErrMessGlobal
+    ENDIF
+    ! TODO: Does the line below need to remove the null char from ErrMessGlobal prior to appending to it?
```

Comment:
I'm under the impress that the transfer doesn't trim off an existing `c_null_char`.

Might have a potential for a buffer overflow here if `ErrMessLocal` is longer than `ErrMsgLen_C - len_trim(RoutineName) - 1` (the `-1` for the `C_Null_Char`)

Reply:
Optionally, you could use the `SetErrStat` on the fortran string to build it up first, then do the transfer.  Will still need space for the `c_null_char` in `ErrMsgLen_C = ErrMsgLen+1`.